build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/GitBranch.cs

```diff
@@ -48,6 +48,12 @@ public override bool Execute ()
 				base.Execute ();
 			}
 
+			// Trim generated dependabot branch names that are too long to produce useful package names
+			var lastSlashIndex = Branch.LastIndexOf ('/');
+			if (Branch.StartsWith ("dependabot") && lastSlashIndex != -1 && Branch.Length > 60) {
+				Branch = Branch.Substring (lastSlashIndex + 1);
+			}
```

Comment:
This looks good. 👍

We can probably just merge this together, but need to check the CI results if the new manifestmerger is OK.